spark-cassandra-connector/src/main/scala/org/apache/spark/sql/cassandra/CassandraConfSetting.scala

```diff
@@ -0,0 +1,190 @@
+    new java.util.HashMap[String, T]())
+  private val keyspaceConf = java.util.Collections.synchronizedMap(
+    new java.util.HashMap[Seq[String], T]())
+  private val tableConf = java.util.Collections.synchronizedMap(
+    new java.util.HashMap[Seq[String], T]())
+
+  /** Add table level configuration settings. Set cluster to None for a single cluster */
+  def addTableConf(
+      table: String,
+      keyspace: String,
+      cluster: Option[String],
+      conf: T): Unit = {
+
+    cluster match {
+      case Some(c) => validateClusterName(c)
+        tableConf.put(Seq(table, keyspace, c), conf)
+      case _ => tableConf.put(Seq(table, keyspace), conf)
+    }
+  }
+
+  /** Add keyspace level configuration settings. Set cluster to None for a single cluster */
+  def addKeyspaceLevelConf(
+      keyspace: String,
+      cluster: Option[String],
+      conf: T) : Unit = {
+
+    cluster match {
+      case Some(c) => validateClusterName(c)
+        keyspaceConf.put(Seq(keyspace, c), conf)
+      case _ => keyspaceConf.put(Seq(keyspace), conf)
+    }
+  }
+
+  /** Add cluster level configuration settings */
+  def addClusterLevelConf(
+      cluster: String,
+      conf: T) : Unit = {
+
+    validateClusterName(cluster)
+    clusterConf.put(cluster, conf)
+  }
+
+  /** Remove table level configuration settings */
+  def removeTableLevelConf(
+      table: String,
+      keyspace: String,
+      cluster: Option[String]) : Unit = {
+
+    cluster match {
+      case Some(c) => validateClusterName(c)
+        tableConf.remove(Seq(table, keyspace, c))
+      case _  => tableConf.remove(Seq(table, keyspace))
+    }
+  }
+
+  /** Remove keyspace level configuration settings */
+  def removeKeyspaceLevelConf(
+      keyspace: String,
+      cluster: Option[String]) : Unit = {
+
+    cluster match {
+      case Some(c) => validateClusterName(c)
+        keyspaceConf.remove(Seq(keyspace, c))
+      case _ => keyspaceConf.remove(Seq(keyspace))
+    }
+  }
+
+  /** Remove cluster level configuration settings */
+  def removeClusterLevelConf(
+      cluster: String) : Unit = {
+
+    clusterConf.remove(cluster)
+  }
+
+  /** Get configuration settings by the order of table level, keyspace level, cluster level, default settings */
+  def getConf(
+      table: String,
+      keyspace: String,
+      cluster: Option[String],
+      defaultConf: T): T = {
+
+    cluster match {
+      case Some(c) => validateClusterName(c)
+        Option(tableConf.get(Seq(table, keyspace, c))).getOrElse(
+          Option(keyspaceConf.get(Seq(keyspace, c))).getOrElse(
+            Option(clusterConf.get(c)).getOrElse(defaultConf)))
+      case _       => Option(tableConf.get(Seq(table, keyspace))).getOrElse(
+        Option(keyspaceConf.get(Seq(keyspace))).getOrElse(defaultConf))
+    }
+  }
+}
+
+object CassandraConfSetting {
+
+  def validateClusterName(
+      cluster: String) = {
+
+    if (StringUtils.isEmpty(cluster)) {
```

Comment:
use `require` instead, it does the same
```code
require(!StringUtils.isEmpty(cluster), "cluster name can't be null or empty")
```
Also, checking for nullness is superfluous in Scala, because you should not use null anywhere besides the code dealing with legacy Java APIs.